package.json

```diff
@@ -4,17 +4,17 @@
   "version": "0.0.0",
   "engines": {
     "yarn": ">=1.7.0 <2",
-    "node": ">=16.14.0 <17"
+    "node": ">=16"
```

Comment:
Curious: the Developing.md was updated to say developers should use >= 18.17 and < 21. But this line says >= 16. Why that discrepancy?

Reply:
While we generally recommend the version range that you've mentioned, there is no technical reason for limiting it to that range. Though we know that < 16.0.0 doesn't work due to native dependencies. 18.17.0 is specifically the recommended minimum version because that's the version that our Electron version uses.

Reply:
We recommend a certain version range in our documentation as we try to make sure it works for said range.

This `engines.node` field will have Yarn stop the build if you use anything outside the defined range, and I might have left it lax so that it's more convenient when switching between node versions for development purposes.

<details>
  <summary>Most other NPM projects -including our deps- are also pretty lax with this range (click to expand).</summary> 

  - https://github.com/websockets/ws/blob/d8dd4852b81982fc0a6d633673968dff90985000/package.json#L32
  - https://github.com/socketio/socket.io/blob/9a2a83fdd42faa840d4f11fd223349e5d8e4d52c/package.json#L92
  - https://github.com/jprichardson/node-fs-extra/blob/a277cbbdf630424482b1e7418749ccaac0db98fe/package.json#L6
  - https://github.com/RogerDeng/drivelist/blob/6b79bf74a40e19bc69a561b978067807060192c4/package.json#L36
  - https://github.com/facebook/react/blob/faa65778b7c277e0020b8bb1fc172e11a863c0b7/packages/react/package.json#L38
  - https://github.com/Axosoft/nsfw/blob/13a84d97c0d778535bbc3baebcc6dc1216a26325/package.json#L22
  - https://github.com/webpack/webpack/blob/87660921808566ef3b8796f8df61bd79fc026108/package.json#L113
  - etc
</details>

But we could move the lower bound along the minimum recommended version.

At the very minimum we need to update this range based on the kind of constructs we use from NodeJS: we can't use new Node 18 syntax and claim that we require anything below Node 18 to run our code base. Although syntax-wise we're dependent on our TypeScript compilation target: No matter what TS syntax we use `tsc` will "dumb it down" for us in the output JS based on our configs. [We're currently targeting ES2019](https://github.com/eclipse-theia/theia/blob/dcfff8326ed4272710e51665a52782a2a2ba2a5e/configs/base.tsconfig.json#L19) and it seems like [Node 16 supports it very well](https://node.green/#ES2019).

Reply:
In the long run it's in everyone's best interest to always use the latest Node LTS, but some might still lag behind so the range can accommodate for it somewhat.

Now I remember: Had I updated the range to `>=18` for that release then all users of the framework would have had build issues because they most likely were still running on Node 16 at that point. So instead of breaking everyone on release I kept it that way so people can update when they're ready. It's not like the project will stop working on Node 16 right away :)